zerver/tests/test_invite.py

```diff
@@ -997,6 +999,47 @@ def test_multi_user_invite(self) -> None:
             ]
         )
 
+    def test_direct_notification_for_accepted_invitation(self) -> None:
+        """
+        New test to check if notification-bot message is sent to the inviter
+        when send_notification is false.
+        """
+        self.login("hamlet")
+
+        invitee = self.nonreg_email("alice")
+        self.assert_json_success(self.invite(invitee, ["Denmark"], False))
+        self.assertTrue(find_key_by_email(invitee))
+        self.submit_reg_form_for_user(invitee, "password")
+
+        invitee_profile = self.nonreg_user("alice")
+        self.assertFalse(invitee_profile.is_realm_admin)
+
+        last_invite_message = Message.objects.all().order_by("-id")[2]
```

Comment:
A better way would be to filter out the messages to find the last message received by the user who created the invite. You can `filter` on the `recipient_id` field.

That way the message to check here would not depend on how many messages are sent to the new invited user.